src/ceph-disk/ceph_disk/main.py

```diff
@@ -2123,20 +2179,101 @@ def __init__(self, args):
         super(PrepareBluestoreBlock, self).__init__(args)
 
     def get_space_size(self):
-        return 0  # get as much space as possible
+        block_size = get_conf(
+            cluster = self.args.cluster,
+            variable = 'bluestore_block_size',
+        )
 
-    def desired_partition_number(self):
-        if self.args.block == self.args.data:
-            num = 2
+        if block_size is None:
+            return 0  # get as much space as possible
         else:
-            num = 0
+            return int(block_size) / 1048576 # MB
+
+    def desired_partition_number(self):
```

Comment:
Although the logic to assign partition number look good, it's a little tricky to read (and maintain). What about setting a fixed partition number for block (2) block.db (3) block.wal (4) ?